chapter_1/2_kcenters.tex

```diff
@@ -67,3 +82,175 @@ \section{The \emph{k}-centers problem}
 better than 2-optimality? Is this the better algorithm to solve the
 problem? 
+
+%%%%%% Gihyen Eom and Gaozhan Wang
+
+\section{The Asymmetric \emph{k}-centers Problem}
+
+We have thus covered the \emph{symmetric} $k$-centers problem. However, we can further generalize the definition of distance. In this section we discuss the \emph{asymmetric} $k$-centers problem and provide an $O(\log^*k)$ approximation.
+
+\begin{definition}
+A distance function $\rho$ is a \emph{metric} if for all $u,v,w$
+\begin{enumerate}
+    \item $\rho(u,v)\geq0$
+    \item $\rho(u,v)=0\iff u=v$
+    \item $\rho(u,w)\leq\rho(u,v)+\rho(v,w)$, and
+    \item $\rho(u,v)=\rho(v,u)$
+\end{enumerate}
+If $\rho$ does not satisfy 4., then it is an \emph{asymmetric} metric (distance function).
+\end{definition}
+
+An example of this is the problem of building fire stations. In general, a street may not necessarily be symmetrically distanced. There are some one-way streets and some streets where you may only make left/right turns. Therefore, using the asymmetric $k$-centers problem is a better solution. Consider the following problem.
+
+\begin{enumerate}
+\item \underline{Input}: a set $V$ of $n$ points, an integer $k\in\mathbb{Z}_{>0}$, and a matrix $D$ specifying the distance function $d:V\times V\rightarrow\mathbb{R}_+\bigcup\{\infty\}$, where $d$ follows the first three criteria above
+\item \underline{Output}: $C\subset V$ with $\vert C \vert \leq k$, is a solution to the $k-center$ problem.
+\item \underline{Goal}: minimize the cost of $C$ where: $cost(C):=max_{v\in V}d(v, C)$ In other words, to find the minimum covering radius $R^\star$ ,such that every point is within $R^\star$ of the set $C$.
+\end{enumerate}
+For $R \geq 0$, we define the graph $G_R=(V,E_R)$, where $E_R = {(u,v):d(u,v)\leq R}.$ The essential connection here is that there exist $k$ center that cover all of $G_R$ iff $R \geq R^\star$. Thus we can binary search for the optimal radius $R^\star$ and work in $G_{R^\star}$. Further, because traversing each edge in the graph corresponds to moving at most $R^\star$ in the original space, finding an $i-cover$ in this graph will yield an $i-approximation$ in the original space.\\\\
+For $y \in \mathbb{R}^S$, we use y(S) to denote $\sum_{v \in S} y_v$, for a function $g$, let $g^(i) $denote the function iterated $i$ times. Finally, define $log^\star x= min\{i:log^{(i)}x\leq \frac 32\}$. Then we can describe a polynomial time relaxed decision procedure AKC, which takes as input an asymmetric $k-center$ instance and a guess at the optimal radius $R$, and either outputs a solution of value $O(Rlog^\star k) $or correctly reports that$R < R\star$. Since $R^\star = d(u,v)$ for some $u,v \in V$, there are only $O(n^2)$ different possible values for $R^\star $. Thus is we binary search on $R$, using at most $O(log n)$ calls to AKC to yield a solution of value $O(Rlog^\star k)$ for some $R\leq R^\star$, and this gives our $\O(log^\star k)-approximation$  algorithm.\\\\
```

Comment:
Can you quickly replace all the log to \log. Should be easy with a simple ctrl+f.

Reply:
Same for min (replace by \min)